packages/stack/src/vendor/views/KeyboardManager.tsx

```diff
@@ -56,10 +56,10 @@ export default class KeyboardManager extends React.Component<Props> {
 
     const input = this.previouslyFocusedTextInput;
 
-    if (Platform.OS === 'android') {
-      Keyboard.dismiss();
-    } else if (input) {
+    if (closing && input) {
       TextInput.State.blurTextInput(input);
```

Comment:
Hey, thanks for the PR, but it seems confusing as to why there's need for separate logic for `closing`. When a screen is closing, it's going to unmount and that should remove focus from the input anyway and hide the keyboard without having to blur it manually.

So I'm not really sure if the issue here is with the keyboard handling. Maybe your view is not unmounting somehow and it's a different issue? Can you share a repro for this?

Reply:
@satya164 
it's there to resolve this issue - #6918
Which happens when the screen closes, when you close the screen and right away focus on the input on the previous screen, you'll get the input blur and focus back

Reply:
I just want to understand the logic behind all the changes. Without understanding the logic, it's easy to break it in future updates and impossible to make any kind of changes. It may fix the issue, but how? Can you add some more description of the behaviour you observe so I can better understand what exactly is happening?

Reply:
@satya164 
Sure, the current behavior is the following: 
Preconditions: you have 2 screens available. On the 1st screen, you have a TextInput, on the second one you don't need anything at all.
1. You open the second screen
2. You click on go back btn
3. You click on input right away while the transition is still in progress
4  The input get focused, then it blurs right away.
If you focus the input second time - it will work just fine, also if you focus after the transition it will also work fine

To resolve this issue, you added the following code: 
`  if (input) {
      TextInput.State.blurTextInput(input);...
`
But the thing is it broke the behavior for the opening of the screen, which caused this issue #7875 

So keeping all this in mind we have to:
1. Dismiss the keyboard on the transition to the new screen.
2. Only unfocus the previously focused inputs if we going back from the screen to allow people immediately focus when they see the previous screen.

The last PR we made for both of these issues solved it only partially, while this one is supposed to address it on both platforms in the right way

Reply:
@satya164  did I miss something important for you here, or this is enough? :)

Reply:
> @satya164
> Sure, the current behavior is the following:
> Preconditions: you have 2 screens available. On the 1st screen, you have a TextInput, on the second one you don't need anything at all.
> 
> 1. You open the second screen
> 2. You click on go back btn
> 3. You click on input right away while the transition is still in progress
>    4  The input get focused, then it blurs right away.
>    If you focus the input second time - it will work just fine, also if you focus after the transition it will also work fine
> 
> To resolve this issue, you added the following code:
> ` if (input) { TextInput.State.blurTextInput(input);...`
> But the thing is it broke the behavior for the opening of the screen, which caused this issue #7875
> 
> So keeping all this in mind we have to:
> 
> 1. Dismiss the keyboard on the transition to the new screen.
> 2. Only unfocus the previously focused inputs if we going back from the screen to allow people immediately focus when they see the previous screen.
> 
> The last PR we made for both of these issues solved it only partially, while this one is supposed to address it on both platforms in the right way

i have this issue did you fixed it ?